src/test/lib/storage/fsst_segment_test.cpp

```diff
@@ -0,0 +1,173 @@
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base_test.hpp"
+
+#include "all_type_variant.hpp"
+#include "storage/chunk.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "storage/fsst_segment.hpp"
+#include "storage/fsst_segment/fsst_encoder.hpp"
+#include "storage/segment_encoding_utils.hpp"
+#include "storage/value_segment.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+// A simple polymorphic memory resource that tracks how much memory was allocated
+class FSSTSimpleTrackingMemoryResource : public boost::container::pmr::memory_resource {
+ public:
+  size_t allocated{0};
```

Comment:
auto :)

Reply:
allocated is a member variable here and thus can't be auto, except we would make it const as well, but it cant be const in this case.